src/gui/TorchMenu.cxx

```diff
@@ -0,0 +1,190 @@
+void TorchMenu::handle_back(Urho3D::StringHash, Urho3D::VariantMap&) { settings_menu->SetVisible(false); }
+void TorchMenu::handle_settings(Urho3D::StringHash, Urho3D::VariantMap&) { settings_menu->SetVisible(true); }
+
+void TorchMenu::handle_reset(Urho3D::StringHash, Urho3D::VariantMap&) {
+    background->SetVisible(true);
+    emu_state = 0;
+    auto* load_button = main_gui->GetChild("load_button", true);
+    load_button->SetEnabled(true);
+    load_button->GetChildDynamicCast<Urho3D::Text>(1)->SetText("Load");
+    main_gui->GetChildDynamicCast<Urho3D::Text>("sketch_info", true)->SetText("  - Current Sketch: ...");
+    Urho3D::VariantMap map{};
+    map[inoReset::P_INO_RESET] = true;
+    SendEvent(E_INO_RESET, map);
+}
+
+void TorchMenu::handle_start(Urho3D::StringHash hash, Urho3D::VariantMap& map) {
+    if (emu_state == 0)
+        (this->*load_ino_selector_handlers)(hash, map);
+    else if (emu_state == 1) {
+        auto* input = GetSubsystem<Urho3D::Input>();
+        main_gui->SetVisible(!main_gui->IsVisible());
+        input->SetMouseMode(main_gui->IsVisible() ? Urho3D::MM_ABSOLUTE : Urho3D::MM_RELATIVE);
+    }
+}
+
+void TorchMenu::handle_ino_set(Urho3D::StringHash, Urho3D::VariantMap&) {
+    auto path = get_selector_path();
+    selector.Reset();
+    ui->GetRoot()->GetChildDynamicCast<Urho3D::Text>("sketch_info", true)->SetText(("  - Current Sketch: " + path.filename().generic_string()).c_str());
+    ino_path = path;
+    send_compile_event();
+}
+
+void TorchMenu::handle_smce_home_set(Urho3D::StringHash, Urho3D::VariantMap&) {
+    auto path = get_selector_path().parent_path();
+    auto* butt = ui->GetRoot()->GetChild("smce_home_set_button", true);
+    butt->GetChildDynamicCast<Urho3D::Text>(1)->SetText((" " + path.generic_string()).c_str());
+    smce_home = path;
+    selector.Reset();
+    send_compile_event();
+}
+
+void TorchMenu::handle_veh_conf_set(Urho3D::StringHash, Urho3D::VariantMap&) {
+    auto path = get_selector_path();
+    auto* child0 = ui->GetRoot()->GetChildDynamicCast<Urho3D::Text>("veh_conf_info0", true);
+    child0->SetText((" " + path.generic_string()).c_str());
+    auto* child1 = ui->GetRoot()->GetChildDynamicCast<Urho3D::Text>("veh_conf_info1", true);
+    child1->SetText(("  - Vehicle Conf: " + path.filename().generic_string()).c_str());
+    veh_conf = path;
+    selector.Reset();
+    send_compile_event();
+}
+
+void TorchMenu::handle_board_conf_set(Urho3D::StringHash, Urho3D::VariantMap&) {
+    auto path = get_selector_path();
+    auto* child0 = ui->GetRoot()->GetChildDynamicCast<Urho3D::Text>("board_conf_info0", true);
+    child0->SetText((" " + path.generic_string()).c_str());
+    auto* child1 = ui->GetRoot()->GetChildDynamicCast<Urho3D::Text>("board_conf_info1", true);
+    child1->SetText(("  - Board Conf: " + path.filename().generic_string()).c_str());
+    board_conf = path;
+    selector.Reset();
+    send_compile_event();
+}
```

Comment:
Code duping is hard here, need to fix by making things a bit more generic with ui tags